sdk/storage_blobs/src/options/mod.rs

```diff
@@ -48,6 +53,17 @@ request_query!(
     /// This type could also be a DateTime but the docs clearly states to treat is as opaque so we do not convert it in any way.
     ///
     /// See: <https://docs.microsoft.com/rest/api/storageservices/get-blob>"]
+    #[derive(PartialEq, Eq, Serialize, Deserialize)]
     Snapshot,
     "snapshot"
 );
+
+impl FromStr for Snapshot {
```

Comment:
What is this needed for? `FromStr` is most oftened used when something is parsed, but we're not really parsing here - we accept any and all strings...

Reply:
This is due to the type signature on `get_optional_as`: https://github.com/Azure/azure-sdk-for-rust/blob/27e1dd45936e77c87cc7bb96bf7b4a7b994a77a7/sdk/core/src/headers/mod.rs#L87-L93 

which is used here: https://github.com/Azure/azure-sdk-for-rust/pull/1008/files#diff-bdeb8cd7bab3372df4aa2feb49411e6cbaf978c135e2dc43330aff4ee30ed32fR233. There are other functions that could be used like `get_optional_str` but then you would have to convert that to the snapshot type via `into`. I am not sure which is better.